src/components/Questionnaire/QuestionnaireForm.tsx

```diff
@@ -687,15 +686,13 @@ export function QuestionnaireForm({
             encounter: encounterId,
             patient: patientId,
             results: validResponses
-              .filter((response) =>
-                isQuestionEnabled(
-                  findQuestionById(
-                    form.questionnaire.questions,
-                    response.question_id,
-                  ) as Question,
-                  form.responses,
-                ),
-              )
+              .filter((response) => {
+                const question = findQuestionById(
+                  form.questionnaire.questions,
+                  response.question_id,
+                ) as Question;
+                return question?._is_enabled ?? true;
+              })
```

Comment:
Can't we compute this instead of caching 🤔 
(Not a suggestion for change, just a question)